# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: 04d9dba72f22
# Full Hash: 04d9dba72f221b5808f077e3aac5522147008438
# Author: Jim Porter <jporter@mozilla.com>
# Date: 2019-06-04 21:42:40
# Regressor Bug: 1552017
# File Overlap Count: 1
# Description:
#   Bug 1552017 - Expand the kinds of URLs that can cancel content JS when navigating; r=smaug
#   
#   This patch makes several changes to the kinds of URLs where we can cancel
#   content JS when navigating between them:
#   
# ==============================================================================

diff -r 25b1ee5ee458 -r 04d9dba72f22 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Tue Jun 04 16:44:29 2019 +0000
+++ b/dom/ipc/BrowserParent.h	Tue Jun 04 16:19:27 2019 +0000
@@ -189,6 +189,13 @@
    */
   bool IsDestroyed() const { return mIsDestroyed; }
 
+  /**
+   * Returns whether we're in the process of creating a new window (from
+   * window.open). If so, LoadURL calls are being skipped until everything is
+   * set up. For further details, see `mCreatingWindow` below.
+   */
+  bool CreatingWindow() const { return mCreatingWindow; }
+
   /*
    * Visit each BrowserParent in the tree formed by PBrowser and
    * PBrowserBridge, including `this`.